### ISSUE TITLE
[Tracking] Current VB work (Refactoring and adding tests)

### ISSUE THREAD

**Issue by KlausLoeffelmann** [dotnet/winforms#12230]
[Draft]

This will be the tracking issues for all the ongoing VB work that we want to accomplish in the .NET 10-time frame.

- [ ] https://github.com/dotnet/winforms/pull/12226 
(Refactoring out large code into 7 logical partial classes/code files)
- [ ] https://github.com/dotnet/winforms/pull/12221 
(Adding tests for VB App Framework Download tests)
- [ ] https://github.com/dotnet/winforms/pull/12227 
(Adding tests for VB App Framework Clipboard class)

**Comment by KlausLoeffelmann** [comment]
@paul1956:

Could you please list your suggested PR order as I started it in an answer to this comment?
I will then take a look as soon as I get to it and paste it into this tracking issue.
Let's then tackle this step by step.

Please do not create another PR, before we have this sorted out! 😸

**Comment by paul1956** [comment]
The first 5 PR’s build on each other I learned not to do that after it was done. The critical issue is to fix #9807, followed by adding tests to the VB runtime. Everything else is general cleanup


#11863 Add tests for VB runtime REVIEW 1ST
Includes the first 3 PR’s below and many more tests, plus a download servers for testing, and some style fixes from reviews (it should be closed and the remaining changes put into small PR’s like at bottom.

                #12227 Add Clipboard Proxy Tests from PR #11863 
                #12226 Split Network.VB into 7 Files  from #11863
                #12221 Extract only Network Download tests from #11863 

#11867 Fix issue#9807 Replaces #9867 REVIEW (2nd) AFTER #11863 DRAFT 
Changes 6 files from #11863 but includes all the code to replace WebClient for Download, optionally expose Async Download, upload is possible but I have no approved way to test. This also includes resources that will need translating they are private to WebClient.

#12118 Shorten long lines in VB REVIEW (3rd) AFTER fix issue#9807 DRAFT 
Purely cleanup no logic changes. This shortens line over 120-130 characters in VB

#12139 Shorten long inline comments REVIEW (4th) DRAFT (purely cleanup no logic changes)
Purely C# cleanup no logic changes

#12148 Improve xml comments review 5th DRAFT #12139 
This shortens comments that are over 120 characters long, this is all C#

All the PR’s below are standalone and small based on Main.

#12141 Add XML Comments related to FileSystemProxy and SpecialDirectoriesProxy 
Adds missing XML comments for VB Public functions.

#12139 shortens comments over 120-130 characters in C#
Purely C# cleanup no logic changes

#12148 improves XML comment by adding see cref’s (in C# and VB)
Purely cleanup no logic changes

**Comment by paul1956** [comment]
This is the complete list of tests and test utilities that I think need to be added to the VB Code, some have already been merged.

The ones in red have already been merged. The TestData Directory gets added to over time as new tests are added.

![image](https://github.com/user-attachments/assets/ee9b027b-d0ff-43f7-b01b-d363ec15baf2)

**Comment by KlausLoeffelmann** [comment]
OK, that helps.
I'll go through those soon, but I think the first one will be the one I have in the list already "to get ramped up" so to say. Nothing for you to do, right now, would like to ask for patience. I'll keep you posted.

**Comment by paul1956** [comment]
@lonitra a few more Pr’s to close issue. They are all listed here. #11863 was split in several pieces to simplify review. 

Thanks

**Comment by lonitra** [comment]
Apologies. Looks like merging PR automatically closed this. I've reopened.

**Comment by paul1956** [comment]
@lonitra should I manually edit this issue to delete the merged issues or somehow mark them closed/merged?

**Comment by adegeo** [comment]
Here are some things I noticed about VB while working on the Winforms Migration article:

- Creating a project in VS, the `HighDpiMode` is missing in `Application.myapp` until you set it via project properties.
- Changing **Application Framework** > **Windows Visual Style** in project properties doesn't apply on next compile. I have run the project twice to get it working.
- In **Solution Explorer**, double-clicking on **My Project** opens the project settings page, but fails to navigate to the **Application Framework** section.
